### PR TITLE
PowerPC VLE asm/anal

### PR DESCRIPTION
**Pull request by wargio**

### CODE REVIEW

File: libr/anal/p/anal_ppc_vle.c [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+	.license = "LGPL3",
+	.name = "ppc.vle",
+	.arch = "ppc",
+	.bits = 32,
+	.op = &vle_anop,
+	.set_reg_profile = &set_reg_profile,
+};
+
+#ifndef CORELIB
+struct r_lib_struct_t radare_plugin = {
```
Comment (radare) [inline libr/anal/p/anal_ppc_vle.c]: use RLibStruct

**Comment by radare** [comment]
is VLE an extension or a different CPU?

**Comment by wargio** [comment]
VLE is special subset of instructions for some PPC SoC.
All the VLE sections are mapped in a different way.
There are some flags into the ELF headers to enable this.
```c
#define SHF_PPC_VLE 0x10000000     /* section header flag */ 
#define PF_PPC_VLE 0x10000000      /* program header flag */
```

**Comment by wargio** [comment]
it's just used to decrease the binary size.

**Comment by wargio** [comment]
ffs

**Comment by radare** [comment]
What about using this disassembler via asm.cpu or asm.bits instead of asm.arch? The pr lgtm but maybe we can avoid creating a new plugin

**Comment by radare** [comment]
But its clear that this is not capstone or gnu :p

**Comment by wargio** [comment]
capstone doesn't support it.
GNU does, but not the ported version of r2.
there are too many cpus using it, on `objdump` you will use `powerpc:vle` as arch.

**Comment by radare** [comment]
my suggestion is to use this disassembler from the gnu and capstone disassemblers when asm.cpu=vle and asm.arch=ppc or ppc.gnu.

another idea would be to upgrade the gnu one and submit this support to capstone instead of having another disassembler plugin. we can have this plugin in extras and make it available via r2pm without any problem, but i would like to not introduce more plugins in core than the ones we really have/need.

**Comment by wargio** [comment]
i don't know how easy could be porting the gnu one.
for capstone, LLVM doesn't support it. so a patch could be done, but i have to read the full src of capstone

**Comment by radare** [comment]
Porting any arch from gnu takes me like 30m. Implementing it in capstone will probably be harder, but sounds like the eay to go to avoid maintaining our own custom disassemblers.

So for now i would go into defining asm.cpu=vle to use your code for both gnu and cs and then i'll merge. After this we can think in updating the ppc disassembler from gnu binutils and eventually implement yhe same thing in capstone, or at least fill an issue in their repobwcause more ppl will be interested in that

**Comment by wargio** [comment]
ok, then i'll have to understand how the asm.cpu thing works..
for the VLE GNU, i don't know which version works.
for the VLE CS, there is already an open issue on the capstone page: https://github.com/aquynh/capstone/issues/560

**Comment by radare** [comment]
ping. i would like to merge this asap

**Comment by wargio** [comment]
Sorry, busy day. I'll try to push this for tomorrow

**Comment by wargio** [comment]
Ok, i think i'll close this and open a new one with just the CPU thing. @radare

**Pull request closed without merging**